### ISSUE TITLE
match_header: make `value` an Option<&str>

### ISSUE THREAD

**Issue by lucab** [lipanski/mockito#11]
I'd like to request/suggest a parameter type change for `match_header()`.
In particular, I would like to request the `value` parameter to be optional (`Option<&str>`).

The usecase is a test I want to write where I'm interested in checking just the presence of a specific header, while I'm not interested in matching the value (which is not always known by the test runner).

**Comment by lipanski** [comment]
this was actually on my list. it could be an `Option`, but I was originally thinking of a trait that could be implemented for `&str` and regular expressions (and other types in the future).

I'll see what comes out and let you know.

**Comment by lipanski** [comment]
@lucab this was just released with version 0.3.0 - check the details here: http://lipanski.github.io/mockito/generated/mockito/index.html#other-header-matchers

TL;DR:

```
use mockito::{mock, Matcher};

mock("GET", "/").match_header("content-type", Matcher::Any).create();
```

**Comment by lucab** [comment]
Great, I'll give it a try. Looking at the docs I think it matches what I'd need, thanks a lot!

For reference, this is where I'd plan to use it: https://github.com/camallo/dkregistry-rs/blob/master/tests/mock/base_client.rs#L40
